internal/ingress/controller/store/store.go

```diff
@@ -539,37 +544,37 @@ func (s *k8sStore) updateSecretIngressMap(ing *extensions.Ingress) {
 		"auth-tls-secret",
 	}
 	for _, ann := range secretAnnotations {
-		secrKey, err := objectRefAnnotationNsKey(ann, ing)
+		annValue, err := parser.GetStringAnnotation(ann, ing)
 		if err != nil && !errors.IsMissingAnnotations(err) {
-			glog.Errorf("error reading secret reference in annotation %q: %s", ann, err)
-			continue
+			glog.Errorf("error parsing annotation %q: %s", ann, err)
 		}
-		if secrKey != "" {
-			refSecrets = append(refSecrets, secrKey)
+		if annValue != "" {
+			secrKey, err := objectRefNamespaceKey(annValue, ing)
+			if err != nil {
+				glog.Errorf("error reading secret reference in annotation %q: %s", ann, err)
+				continue
+			}
+			if secrKey != "" {
+				refSecrets = append(refSecrets, secrKey)
+			}
 		}
 	}
 
 	// populate map with all secret references
 	s.secretIngressMap.Insert(key, refSecrets...)
 }
 
-// objectRefAnnotationNsKey returns an object reference formatted as a
-// 'namespace/name' key from the given annotation name.
-func objectRefAnnotationNsKey(ann string, ing *extensions.Ingress) (string, error) {
-	annValue, err := parser.GetStringAnnotation(ann, ing)
-	if annValue == "" {
-		return "", err
-	}
-
-	secrNs, secrName, err := cache.SplitMetaNamespaceKey(annValue)
+// objectRefNamespaceKey takes an object key in either 'name' or 'namespace/name'
+// form and returns it prepended with the Ingress namespace in the former case.
+func objectRefNamespaceKey(key string, ing *extensions.Ingress) (string, error) {
+	secrNs, secrName, err := cache.SplitMetaNamespaceKey(key)
 	if secrName == "" {
```

Comment:
just wondering why not use 

```
if err != nill {
		return "", err
	}
```

Reply:
It would be redundant.
* If you get an empty string for valid reasons, you return that empty string and err will be nil.
* If you get an empty string due to a parsing error, you return an empty string (null type) and err will be an actual error.
In that particular case compacting 2 cases in one is equivalent.